src/doom/p_setup.c

```diff
@@ -531,6 +531,10 @@ void P_LoadBlockMap (int lump)
 
     lumplen = W_LumpLength(lump);
     count = lumplen / 2;
+    if(count >= 32766)
```

Comment:
Why exactly this number?

Reply:
It's the blockmap limit for doom / 2 (count = blockmap size / 2, and blockmap limit is 65536 in doom, so I put that /2 and checked it against count.

Reply:
Source: https://doomwiki.org/wiki/Blockmap#Compression
and C data type size for short (which is what blockmap is) is 65536

Reply:
Thank you, but this was a rhetorical question. :wink: I'd prefer if magic numbers in source code didn't "fall from the sky". Something like `USHRT_MAX/2` and an accompanying comment would probably be more meaningful.

Reply:
Also, 32766 is the wrong number since 65536 / 2 = 32768. A good example of why Fabian's approach is the better one.